### PR TITLE
Tour overlay mask

### LINKED ISSUE #2197: Build tour overlay component
Spec: https://docs.google.com/document/d/1kfpRShJe0tAA-ZCpUrxeaH8z452y_MqChwJGNk08xQY/edit#heading=h.cylvjjuf8ht2

Prototype: https://jsfiddle.net/bartaz/63y2vxgz/

- [ ] build overlay styles
- [ ] build functionality to position the 'window' anywhere on the page

### PR DESCRIPTION
**Pull request by bartaz**
Fixes #2197 

Adds proof of concept for an overlay component with a mask to highlight an element on the page.

This is a demo code to test the overlay styling. Merged to feature branch (not going to production until tour functionality is finished).

### QA
- ./run or https://snapcraft-io-canonical-web-and-design-pr-2205.run.demo.haus/
- go to home page
- wait for it
- click anywhere on the page to move highlight and tooltip to clicked element

This PR is about dark overlay with mask in it. There may be some issues with tooltip positioning next to window borders.

Screenshot 2019-08-21 at 16 53 31

### CODE REVIEW

File: static/js/publisher/tour/tour.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  return {
+    top,
+    bottom,
+    left,
+    right
+  };
+};
+
+export default function Tour({ targetEl }) {
+  const [rect, setRect] = useState(targetEl ? getRectFromEl(targetEl) : null);
```
Comment (hatched) [inline static/js/publisher/tour/tour.js]: Above this line would be a good place to put a validator that the element exists and is in the dom.

File: static/js/publisher/tour.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import ReactDOM from "react-dom";
+
+import Tour from "./tour/tour";
+
+export default function initTour() {
+  const tour = document.createElement("div");
```
Comment (hatched) [inline static/js/publisher/tour.js]: I'm not a fan of this pattern in a React app but as I understand this will only be temporary.

**Comment by kwm14** [comment]
Design LGTM 👍nice touch with **Skip tour** link #rickastley

**Comment by hughesjordan** [comment]
@bartaz looks awesome, only thing is it get's a bit weird when you resize the window, possible to have it readjust on window resize?

File: static/js/publisher/tour.js (hunk 2 above)
Comment (bartaz) [reply static/js/publisher/tour.js]: Agreed, not a big fan of that as well ;)

**Comment by bartaz** [comment]
@hughesjordan Yes, I think we will need to do that. Will add something.

**Comment by anthonydillon** [comment]
Skip tour, love it! 👍

**Comment by hughesjordan** [comment]
Looking good @bartaz 👍

**Pull request merged**